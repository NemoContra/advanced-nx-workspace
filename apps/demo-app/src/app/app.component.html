<flight-workspace-shared-util-random-animal-image></flight-workspace-shared-util-random-animal-image>
