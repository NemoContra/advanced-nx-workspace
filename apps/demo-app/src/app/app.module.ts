import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { SharedUtilRandomAnimalImageModule } from '@flight-workspace/shared/util/random-animal-image';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    SharedUtilRandomAnimalImageModule.forRoot({ cats: true, dogs: true }),
    RouterModule.forRoot([
      {
        path: '',
        loadChildren: () =>
          import('./animal/animal.module').then((m) => m.AnimalModule),
      },
      {
        path: 'cat',
        loadChildren: () => import('./cat/cat.module').then((m) => m.CatModule),
      },
      {
        path: 'dog',
        loadChildren: () => import('./dog/dog.module').then((m) => m.DogModule),
      },
      {
        path: '',
        component: AppComponent,
      },
      {
        path: '**',
        redirectTo: '',
        pathMatch: 'full',
      },
    ]),
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
