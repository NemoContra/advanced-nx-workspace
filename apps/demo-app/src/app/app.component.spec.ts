import { createComponentFactory } from '@ngneat/spectator/jest';
import { AppComponent } from './app.component';
import { RouterTestingModule } from '@angular/router/testing';

describe('AppComponent', () => {
  const createComponent = createComponentFactory({
    component: AppComponent,
    imports: [RouterTestingModule],
  });

  it('should render', () => {
    const spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
