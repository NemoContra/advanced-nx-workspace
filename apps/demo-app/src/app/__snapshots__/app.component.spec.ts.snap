// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render 1`] = `
<flight-workspace-root
  __ngContext__={[Function LRootView]}
>
  <ul>
    <li>
      <a
        href="/"
        routerlink=""
      >
        Animal
      </a>
    </li>
    <li>
      <a
        href="/cat"
        routerlink="/cat"
      >
        Cat
      </a>
    </li>
    <li>
      <a
        href="/dog"
        routerlink="/dog"
      >
        Dog
      </a>
    </li>
  </ul><router-outlet />
</flight-workspace-root>
`;
